pkg/oam/util/helper.go

```diff
@@ -475,10 +475,39 @@ func PassLabel(parentObj, childObj labelAnnotationObject) {
 // PassLabelAndAnnotation passes through labels and annotation objectMeta from the parent to the child object
 // when annotation or labels has conflicts, the parentObj will override the childObj.
 func PassLabelAndAnnotation(parentObj, childObj labelAnnotationObject) {
```

Comment:
This util function should not have business-specific logic such as filter labels&annots of app.
I would like to suggest you implement this logic [here](https://github.com/oam-dev/kubevela/blob/b582ccc0600dc5cb6c313015ac7431cef3e057e5/pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go#L203-L205). This is where collecting app's lables&annots being passed to workload/trait. Filter lables&annots in that method.

Reply:
Additionally, I made a simple annotation filter [here](https://github.com/oam-dev/kubevela/blob/b582ccc0600dc5cb6c313015ac7431cef3e057e5/pkg/controller/core.oam.dev/v1alpha2/application/assemble/assemble.go#L256). If necessary, you can remove it and merge this logic into yours.

Reply:
The other option is [here ](https://github.com/oam-dev/kubevela/blob/b582ccc0600dc5cb6c313015ac7431cef3e057e5/pkg/controller/core.oam.dev/v1alpha2/application/revision.go#L436-L438)where passing lables&annots to appRevision. This a path how label&annots propagated:  app => appRevision => workload/trait.  So if you need filter these label&annots from appRevision, do it here. It's up to your requirement.